src/execution_tree/primitives/base_primitive.cpp

```diff
@@ -2035,6 +2035,110 @@ namespace phylanx { namespace execution_tree
                 name, codename));
     }
 
+    ///////////////////////////////////////////////////////////////////////////
+    std::int64_t extract_scalar_nonneg_integer_value_strict(
+        primitive_argument_type const& val, std::string const& name,
+        std::string const& codename)
+    {
```

Comment:
Would it be easier to implement this in terms of `extract_scalar_integer_value_strict` and do the non-neg check if that one succeeded?